tensorflow-frameworks/tensorflow-data/src/main/java/org/tensorflow/data/impl/TensorSliceDataset.java

```diff
@@ -0,0 +1,32 @@
+package org.tensorflow.data.impl;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.tensorflow.DataType;
+import org.tensorflow.Operand;
+import org.tensorflow.data.Dataset;
+import org.tensorflow.data.Utils;
+import org.tensorflow.op.Ops;
+
+public class TensorSliceDataset extends Dataset {
+  private org.tensorflow.op.data.TensorSliceDataset dataset;
+
+  public TensorSliceDataset(Ops tf, List<Operand<?>> components, List<DataType<?>> outputTypes) {
+    super(tf, outputTypes,
+        components.stream()
+            .map(c -> Utils.tail(c.asOutput().shape()))
```

Comment:
Why do we need to remove the first dimension of each tensor shapes?

Reply:
The 'outputSize` is supposed to be the size for the returned elements from the dataset. in a standard 2-D `TensorSliceDataset`, for example, it's the "length of each vector" that counts as the element shape, not the "number of vectors"